internal/services/applications/parse/app_role.go

```diff
@@ -76,3 +76,17 @@ func (id AppRoleId) Segments() []resourceids.Segment {
 func (id AppRoleId) String() string {
 	return fmt.Sprintf("App Role (Application ID: %q, Role ID: %q)", id.ApplicationId, id.RoleID)
 }
+
+func (id AppRoleId) FromParseResult(input resourceids.ParseResult) error {
```

Comment:
This'll need to be a pointer:

```suggestion
func (id *AppRoleId) FromParseResult(input resourceids.ParseResult) error {
```